SH4RE/SH4RE/Messages/MessagesInboxView.swift

```diff
@@ -0,0 +1,200 @@
+            .order(by: FirebaseConstants.timestamp)
+            .addSnapshotListener { querySnapshot, error in
+                if let error = error {
+                    self.errorMessage = "Failed to listen for recent messages: \(error)"
+                    print(error)
+                    return
+                }
+                
+                querySnapshot?.documentChanges.forEach({ change in
+                    let docId = change.document.documentID
+                    
+                    if let index = self.recentMessages.firstIndex(where: { rm in
+                        return rm.id == docId
+                    }) {
+                        self.recentMessages.remove(at: index)
+                    }
+                    do {
+                        let data = change.document.data()
+                        let text = data["text"] as? String ?? ""
+                        let name = data["name"] as? String ?? ""
+                        let fromId = data["fromId"] as? String ?? ""
+                        let toId = data["toId"] as? String ?? ""
+                        let timestamp = (data["timestamp"] as? Timestamp)?.dateValue() ?? Date()
+                        let rm = RecentMessage(id: docId, text: text, name:name, fromId: fromId, toId: toId, timestamp: timestamp)
+                        self.recentMessages.insert(rm, at: 0)
+                    }
+                })
+            }
+    }
+    
+    func fetchCurrentUser() {
+        guard let uid = FirebaseManager.shared.auth.currentUser?.uid else {
+            self.errorMessage = "Could not find firebase uid"
+            return
+        }
+        let docRef = Firestore.firestore().collection("User Info").document(uid)
+        docRef.getDocument{ (document, error) in
+            guard error == nil else{
+                print("Error reading document:", error ?? "")
+                return
+            }
+            if let document = document, document.exists {
+                let data = document.data()!
+                let id = document.documentID
+                let name = data["name"] as? String ?? ""
+                FirebaseManager.shared.currentUser = ChatUser(uid:id, name:name)
+            }
+        }
+    }
+    
+    func handleSignOut() {
+        isUserCurrentlyLoggedOut.toggle()
+        try? FirebaseManager.shared.auth.signOut()
+    }
+    
+}
+
+struct MessagesInboxView: View {
+    @Binding var tabSelection: Int
+    @State  var searchQuery: String = ""
+    @EnvironmentObject var currentUser: CurrentUser
+    @ObservedObject var vm = MainMessagesViewModel()
+    @State var shouldNavigateToChatLogView = false
+    var chatLogViewModel = ChatLogViewModel(chatUser: nil)
+    
+    var body: some View {
+
+        ZStack {
+            Color.backgroundGrey.ignoresSafeArea()
+            
+            if (currentUser.isGuest()) {
+                GuestView(tabSelection: $tabSelection).environmentObject(currentUser)
```

Comment:
can you create a nother default view like this if inbox is empty, a simple note on screen that says "Your messages will appear here when you request to rent a listing", then maybe a button that redirects to the search page. Up to you, page just looks weird when it's empty

Reply:
yuh